packages/ember-data/lib/adapters/build-url-mixin.js

```diff
@@ -52,8 +52,11 @@ export default Ember.Mixin.create({
     var url = [];
     var host = get(this, 'host');
     var prefix = this.urlPrefix();
+    var path;
 
-    if (type) { url.push(this.pathForType(type)); }
+    if (type && (path = this.pathForType(type))) {
```

Comment:
Can you take the assignment out of the if statement? I don't think it's needed there.